frame/nomination-pools/src/tests.rs

```diff
@@ -4768,3 +4980,319 @@ mod reward_counter_precision {
 			});
 	}
 }
+
+// NOTE: run this with debug_assertions, but in release mode.
+#[cfg(feature = "fuzz-test")]
+mod fuzz_test {
+	use super::*;
+	use crate::pallet::{Call as PoolsCall, Event as PoolsEvents};
+	use frame_support::traits::UnfilteredDispatchable;
+	use rand::{seq::SliceRandom, thread_rng, Rng};
+	use sp_runtime::{assert_eq_error_rate, Perquintill};
+
+	const ERA: BlockNumber = 1000;
+	const MAX_ED_MULTIPLE: Balance = 10_000;
+	const MIN_ED_MULTIPLE: Balance = 10;
+
+	// not quite elegant, just to make it available in random_signed_origin.
+	const REWARD_AGENT_ACCOUNT: AccountId = 42;
+
+	/// Grab random accounts, either known ones, or new ones.
+	fn random_signed_origin<R: Rng>(rng: &mut R) -> (Origin, AccountId) {
+		let count = PoolMembers::<T>::count();
+		if rng.gen::<bool>() && count > 0 {
+			// take an existing account.
+			let skip = rng.gen_range(0..count as usize);
+
+			// this is tricky: the account might be our reward agent, which we never want to be
+			// randomly chosen here. Try another one, or, if it is only our agent, return a random
+			// one nonetheless.
+			let candidate = PoolMembers::<T>::iter_keys().skip(skip).take(1).next().unwrap();
+			let acc =
+				if candidate == REWARD_AGENT_ACCOUNT { rng.gen::<AccountId>() } else { candidate };
+
+			(Origin::signed(acc), acc)
+		} else {
+			// create a new account
+			let acc = rng.gen::<AccountId>();
+			(Origin::signed(acc), acc)
+		}
+	}
+
+	fn random_ed_multiple<R: Rng>(rng: &mut R) -> Balance {
+		let multiple = rng.gen_range(MIN_ED_MULTIPLE..MAX_ED_MULTIPLE);
+		ExistentialDeposit::get() * multiple
+	}
+
+	fn fund_account<R: Rng>(rng: &mut R, account: &AccountId) {
+		let target_amount = random_ed_multiple(rng);
+		if let Some(top_up) = target_amount.checked_sub(Balances::free_balance(account)) {
+			let _ = Balances::deposit_creating(account, top_up);
+		}
+		assert!(Balances::free_balance(account) >= target_amount);
+	}
+
+	fn random_existing_pool<R: Rng>(mut rng: &mut R) -> Option<PoolId> {
+		BondedPools::<T>::iter_keys().collect::<Vec<_>>().choose(&mut rng).map(|x| *x)
+	}
+
+	fn random_call<R: Rng>(mut rng: &mut R) -> (crate::pallet::Call<T>, Origin) {
+		let op = rng.gen::<u8>();
+		match op % 8 {
```

Comment:
```suggestion
		let op = rng.gen::<usize>();
		let mut max_op_index =
			<crate::pallet::Call<T> as frame_support::dispatch::GetCallName>::get_call_names()
				.len();
		// Exclude set_state, set_metadata, set_configs, update_roles and chill.
		max_op_index -= 5;

		match op % max_op_index {
```
Could be overkill but like this you wont forget to add future extrinsics.

Reply:
bueno